unit-tests/src/org/commcare/android/tests/caselist/EntityListCalloutDataTest.java

```diff
@@ -0,0 +1,189 @@
+package org.commcare.android.tests.caselist;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.widget.ImageButton;
+import android.widget.ListView;
+
+import com.simprints.libsimprints.Identification;
+
+import junit.framework.Assert;
+
+import org.commcare.CommCareApplication;
+import org.commcare.activities.EntitySelectActivity;
+import org.commcare.activities.components.EntitySelectCalloutSetup;
+import org.commcare.adapters.EntityListAdapter;
+import org.commcare.android.CommCareTestRunner;
+import org.commcare.android.util.ActivityLaunchUtils;
+import org.commcare.android.util.TestAppInstaller;
+import org.commcare.android.util.TestUtils;
+import org.commcare.dalvik.BuildConfig;
+import org.commcare.dalvik.R;
+import org.commcare.models.AndroidSessionWrapper;
+import org.commcare.session.CommCareSession;
+import org.commcare.suite.model.Callout;
+import org.commcare.suite.model.Detail;
+import org.commcare.suite.model.EntityDatum;
+import org.commcare.views.EntityView;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.Shadows;
+import org.robolectric.annotation.Config;
+import org.robolectric.shadows.ShadowActivity;
+import org.robolectric.shadows.ShadowListView;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Test attaching callout data to entity select list
+ *
+ * @author Phillip Mates (pmates@dimagi.com)
+ */
+@Config(application = CommCareApplication.class,
+        constants = BuildConfig.class)
+@RunWith(CommCareTestRunner.class)
+public class EntityListCalloutDataTest {
+    private EntitySelectActivity entitySelectActivity;
+    private EntityListAdapter adapter;
+
+    @Before
+    public void setup() {
+        String appProfileResource =
+                "jr://resource/commcare-apps/case_list_lookup/profile.ccpr";
+        TestAppInstaller.initInstallAndLogin(appProfileResource, "test", "123");
+
+        TestUtils.processResourceTransactionIntoAppDb("/commcare-apps/case_list_lookup/restore.xml");
+    }
+
+    @Test
+    public void testAttachCalloutResultToListTest() {
+        launchEntitySelectActivity("m1-f0");
+
+        loadList();
+        // TODO PLM: This fails non-deterministically; haven't figured out how to fix it
```

Comment:
No, https://github.com/dimagi/commcare-odk/pull/1119#discussion_r61756662 fixed it